src/borg/platform/posix.pyx

```diff
@@ -8,3 +13,58 @@ def swidth(s):
         return terminal_width
     else:
         return str_len
+
+
+# only determine the PID and hostname once.
+# for FUSE mounts, we fork a child process that needs to release
+# the lock made by the parent, so it needs to use the same PID for that.
+_pid = os.getpid()
+# XXX this sometimes requires live internet access for issuing a DNS query in the background.
+_hostname = socket.gethostname()
```

Comment:
I don't think there's a way around that API-wise, if the machine is configured that way.